src/libraries/Common/src/System/Reflection/Metadata/TypeNameParserOptions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Reflection.Metadata
+{
+#if SYSTEM_PRIVATE_CORELIB
+    internal
+#else
+    public
+#endif
```

Comment:
```suggestion
#if SYSTEM_REFLECTION_EMIT
    public
#else
    internal
#endif
```
For public types that are included as private in other assemblies, the usual pattern is to place the public under the ifdef. 

Also, we typically put these types under their public assembly directory (ie System.Reflection.Metadata in this case) and not under Common.

One example of many: https://github.com/dotnet/runtime/blob/9aaf6027cae3fe92e5d5903adaddfd1b0dbdb1fa/src/libraries/System.Private.CoreLib/src/System/Index.cs#L18-L22

Reply:
If you don't mind I am going to create a dedicated issue that lists everything that needs to be done in the tooling projects once the SDK version is bumped (to a version that includes the new SRM) and add it there. It simply requires me to make a lot of changes that would need to be reverted very soon (and I am already out of time)

Reply:
The suggestion above is about changing the polarity of the ifdef to follow the usual convention for shared public/internal types. It is valid even if tooling projects are switched to System.Reflection.Metadata.

Reply:
> The suggestion above is about changing the polarity of the ifdef to follow the usual convention for shared public/internal types. It is valid even if tooling projects are switched to System.Reflection.Metadata.

With the current implementation, the public types are included as links in `ILCompiler.TypeSystem`, which is referenced by `ILCompiler.Compiler`. If I change the ifdefs, the `ILCompiler.Compiler` fails to compile. If I include the links to the types in ``ILCompiler.Compiler` it builds fine, but I am not sure if it's not going to fail at runtime (method was expecting type `TypeName` from library `A`, but it got `TypeName` from library `B`)

Reply:
I've created https://github.com/dotnet/runtime/issues/101541 to track it